### PR TITLE
Adding the coc extensions for Apache Camel

### PR DESCRIPTION
**Pull request by Navyashree0923**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    // console.log("Couldn't find Jar File, So downloading the Camel LS Jar")
+    download_camells_Jar(camells_Jar.remote_Path,
+                 camells_Jar.version,
+                 camells_Jar.name,
+                 camells_Jar.suffix
+    )
+    }
+    
+    const serverOptions = {
+        // command: '/usr/bin/java -jar'.concat(' ').concat(jarPath).toString() // run camells
```
Comment (apupier) [inline src/index.ts]: commented line can be removed

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        command: 'java',
+        args: ["-jar", jarPath]
+    }
+    
+    const clientOptions = {
+        documentSelector: ['java', 'xml'] // run camells on java files
```
Comment (apupier) [inline src/index.ts]: not working for yaml files?

File: src/jarDownload.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// https://repo1.maven.org/maven2/com/github/camel-tooling/camel-lsp-server/1.5.0/camel-lsp-server-1.5.0.jar
+//
+
+import * as https from 'https'
+import * as fs from 'fs'
+
+// const urlPrefix = "https://repo1.maven.org/maven2/com/github/camel-tooling/camel-lsp-server/"
+// const url = urlPrefix + '1.5.0' + '/' + 'camel-lsp-server-' + '1.5.0' + '.jar'
+
+// console.log("Complete url:", url)
```
Comment (apupier) [inline src/jarDownload.ts]: commented code to remove

File: src/jarDownload.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export function download_camells_Jar(jar_Url: string, jar_Version: string, jar_Name: string, jarSuffix: string){
+    // console.log("camellsJar:", jar_Url + jar_Version + '/' + jar_Name + jar_Version + '.jar')
```
Comment (apupier) [inline src/jarDownload.ts]: commented code to remove

File: src/jarDownload.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    https.get(remote_Path, 
+         (res) => {
+             const local_Path = '/opt' + '/' + jar_Name + '-' + jar_Version + jarSuffix 
+             const writeStream = fs.createWriteStream(local_Path);
+
+             res.pipe(writeStream)
+             writeStream.on("finish", () => {
+                 writeStream.close()
+                 // console.log("Download Completed")
```
Comment (apupier) [inline src/jarDownload.ts]: commened code to remove

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    local_Path: '/opt',
+    suffix: '.jar'
+    }
+
+    const jarPath = camells_Jar.local_Path + '/' + camells_Jar.name + '-' + camells_Jar.version + camells_Jar.suffix
+    const jarFound = glob.sync(jarPath)
+    
+    // Verify camells is already available on the local system
+    if (! Boolean(jarFound[0])) {
+    // console.log("Couldn't find Jar File, So downloading the Camel LS Jar")
```
Comment (apupier) [inline src/index.ts]: commented code to remove

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      if (!isEnable) {
+        return
+    }
+
+    const camells_Jar = {
+    name: 'camel-lsp-server',
+    version: '1.5.0',
+    remote_Path: 'https://repo1.maven.org/maven2/com/github/camel-tooling/camel-lsp-server/',
+    local_Path: '/opt',
+    suffix: '.jar'
```
Comment (apupier) [inline src/index.ts]: wrong indentation. This should have one more level of indentation.

File: src/index.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+     * 2. Start language server through coc.nvim Library instead of making changes to coc-settings.json
+     *
+     *
+     * Also had few issues storing camells Jar in the github repo along with other regular files. So download camells Jar on-demand and 
+     * removing it from coc-camel/server
+     *                         
+     */
+    const config = workspace.getConfiguration('coc-camel')
+    const isEnable = config.get<boolean>('enable', true)
+      if (!isEnable) {
```
Comment (apupier) [inline src/index.ts]: wrong indentation

File: package.json [hunk 9]
```diff
@@ -0,0 +1,63 @@
+{
+  "name": "coc-camel",
+  "version": "0.0.0-development",
+  "description": "",
```
Comment (apupier) [inline package.json]: would be nice to add a description

File: package.json [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  "author": "npr <npr@redhat.com>",
+  "license": "Apache-2.0",
+  "main": "lib/index.js",
+  "keywords": [
+    "coc.nvim"
+  ],
```
Comment (apupier) [inline package.json]: other Keyword suggestions:
- Apache Camel
- vim

File: README.md [hunk 11]
```diff
@@ -1,2 +1,23 @@
...
+
+`nmap <silent> <C-l> <Plug>(coc-coc-camel-keymap)`
+
+## Lists
+
+`:CocList demo_list`
+
+## License
+
+MIT
```
Comment (apupier) [inline README.md]: License is Apache

File: README.md [hunk 12]
```diff
@@ -1,2 +1,23 @@
...
+
+`:CocInstall coc-camel`
+
+## Keymaps
+
+`nmap <silent> <C-l> <Plug>(coc-coc-camel-keymap)`
+
+## Lists
+
+`:CocList demo_list`
```
Comment (apupier) [inline README.md]: What is the this command doing?

File: src/index.ts [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    
+    const clientOptions = {
+        documentSelector: ['java', 'xml'] // run camells on java files
+    }
+    const client = new LanguageClient('coc-camel', // the id
+                              'coc-camel', // the name of the language server
+                          serverOptions,
+                          clientOptions,
+                             true
```
Comment (apupier) [inline src/index.ts]: wrong indentation on some of the lines

File: README.md [hunk 14]
```diff
@@ -1,2 +1,23 @@
 # coc-camel
-Extension for Vim coc to provide Camel Textual Edition support
+
+
```
Comment (apupier) [inline README.md:5]: Would be nice to include a description of the covered features in the readme. Providing animated image would be nice too.
it is also possible to handle it in another iteration (and so create an issue for it, then provide a different Pull Request)

**Review by apupier — CHANGES_REQUESTED** [review]

**Comment by Navyashree0923** [comment]
Thank you for the feedback. I have updated the PR with the modifications

**Comment by apupier** [comment]
it remains two of my comments not adressed:
- https://github.com/camel-tooling/coc-camel/pull/2#discussion_r913749237
- https://github.com/camel-tooling/coc-camel/pull/2#discussion_r913749901

**Review by apupier — CHANGES_REQUESTED** [review]
I'm unable to build locally.
What are the steps to do the build? Would be nice to indicate them in a Contributing.md file.

Edit: got a build with the following steps:
```
npm install esbuild
yarn
```
Is it the right ones?

**Comment by Navyashree0923** [comment]
> I'm unable to build locally. What are the steps to do the build? Would be nice to indicate them in a Contributing.md file.
> 
> Edit: got a build with the following steps:
> 
> ```
> npm install esbuild
> yarn
> ```
> 
> Is it the right ones?

Yeah these are correct. I have added a new session in the readme on how to contribute instead of creating a new contribute.md file.
Thanks for the feedback

**Comment by Navyashree0923** [comment]
I have also added a `.gitignore` to avoid tracking of files like `node_modules`, `lib`, etc

**Review by apupier — APPROVED** [review]
Sounds great!

**Pull request merged**